staging/src/k8s.io/apiserver/pkg/endpoints/handlers/delete_test.go

```diff
@@ -215,6 +222,530 @@ func TestDeleteCollection(t *testing.T) {
 	}
 }
 
+func TestDeleteResourceWithUnsafeDeletionFlow(t *testing.T) {
+	tests := []struct {
+		name           string
+		featureEnabled bool
+		// whether the registry implements rest.CorruptObjectDeleterProvider, and provides an unsafe deleter:
+		//  nil: it does not implement the rest.CorruptObjectDeleterProvider interface
+		//  true: it implements CorruptObjectDeleterProvider, and returns a valid unsafe deleter
+		//  false: it implements CorruptObjectDeleterProvider, but returns nil when asked for an unsafe deleter
+		registryHasUnsafeDeleter *bool
+		// what the user passes in the delete options for ignoreStoreReadErrorWithClusterBreakingPotential
+		ignoreReadErr      *bool
+		authorizer         authorizer.Authorizer
+		normalFlowObserved *deletetionFlowObserved // records what the normal deletion flow observes
+		unsafeFlowObserved *deletetionFlowObserved // records what the unsafe deletion flow observes
+		// want
+		normalDeletionFlowWant *deletetionFlowObserved // what the normal deletion flow should observe
+		unsafeDeletionFlowWant *deletetionFlowObserved // what the unsafe deletion flow should observe
+		statusCodeWant         int                     // what the delete handler should return
+		errshouldContain       string                  // the expected error in the response body
+		unsafeAnnotationWant   bool                    // whether the unsafe audit annotation should be added
+	}{
+		{
+			name:                     "feature disabled, registry does not implement CorruptObjectDeleterProvider, ignore is false, should invoke the normal deletion flow",
+			featureEnabled:           false,
+			registryHasUnsafeDeleter: nil,
+			ignoreReadErr:            ptr.To[bool](false),
```

Comment:
Dunno what the best practice is in k8s, but my linter suggests this:

```suggestion
			ignoreReadErr:            ptr.To(false),
```